From e5e754f59e6a847c9573fa9f226fd1266783b99d Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:25 -0400
Subject: [fs] btrfs: remove redundant null check in btrfs_dentry_release()

Message-id: <1407179109-13714-319-git-send-email-sandeen@redhat.com>
Patchwork-id: 87280
O-Subject: [RHEL7.1 PATCH 318/422] btrfs: remove redundant null check in btrfs_dentry_release()
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Daeseok Youn <daeseok.youn@gmail.com>

commit 944a4515b27302ac42d5964b7400c72f4d692782
Author: Daeseok Youn <daeseok.youn@gmail.com>
Date:   Mon Apr 14 15:37:02 2014 +0900

    btrfs: remove redundant null check in btrfs_dentry_release()

    It doesn't need to check NULL for kfree()

    Signed-off-by: Daeseok Youn <daeseok.youn@gmail.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/inode.c b/fs/btrfs/inode.c
index f340e8d..9d4db28 100644
--- a/fs/btrfs/inode.c
+++ b/fs/btrfs/inode.c
@@ -5164,8 +5164,7 @@ static int btrfs_dentry_delete(const struct dentry *dentry)
 
 static void btrfs_dentry_release(struct dentry *dentry)
 {
- if (dentry->d_fsdata)
-  kfree(dentry->d_fsdata);
+ kfree(dentry->d_fsdata);
 }
 
 static struct dentry *btrfs_lookup(struct inode *dir, struct dentry *dentry,
-- 
1.7.1